### PR TITLE
Fix - Cell Reordering in IGListBindingSectionController (viewModels array not being updated)

### PR DESCRIPTION
**Pull request by d3mueller**
- See #1262 (cell reordering using a binding section controller caused problems)

## Changes in this pull request
- Added method override in `IGListBindingSectionController` to update the internal `viewModels` array after a cell has been moved
- Subclasses of IGListBindingSectionController require to call super

Issue fixed: #1262 

I've added a test for this change. Is this enough? Or should I add more corner cases or something?

### Checklist

Some tests fail, but they do seem unrelated. I don't know if this is normal. I haven't changed anything regarding those failing tests.

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: Source/IGListBindingSectionController.h [hunk 1]
```diff
@@ -77,6 +77,8 @@ NS_SWIFT_NAME(ListBindingSectionController)
  */
 - (void)updateAnimated:(BOOL)animated completion:(nullable void (^)(BOOL updated))completion;
 
+- (void)moveObjectFromIndex:(NSInteger)sourceIndex toIndex:(NSInteger)destinationIndex NS_REQUIRES_SUPER;
```
Comment (rnystrom) [inline Source/IGListBindingSectionController.h]: Can you add some doc headers? I think copy/paste the superclass ones but add a note about why super is required?

File: Tests/IGListBindingSectionControllerTests.m [hunk 2]
```diff
@@ -395,5 +395,21 @@ - (void)test_whenUpdating_withMutableArrayObject_thatViewModelsDontMutate {
...
+    
+    IGTestDiffingSectionController *section = [self.adapter sectionControllerForObject:self.dataSource.objects.firstObject];
+    
+    [section moveObjectFromIndex:0 toIndex:2];
+    XCTAssertEqual([section.viewModels firstObject], @"seven");
+    XCTAssertEqual([section.viewModels lastObject], @7);
+    
+    [section moveObjectFromIndex:2 toIndex:1];
+    XCTAssertEqual([section.viewModels objectAtIndex: 1], @7);
+    XCTAssertEqual([section.viewModels lastObject], @20);
```
Comment (rnystrom) [inline Tests/IGListBindingSectionControllerTests.m]: Just confirming, this fails w/out the patch applied?

File: Source/IGListBindingSectionController.h (hunk 1 above)
Comment (d3mueller) [reply Source/IGListBindingSectionController.h]: Sure, I committed the changes.

File: Tests/IGListBindingSectionControllerTests.m (hunk 2 above)
Comment (d3mueller) [reply Tests/IGListBindingSectionControllerTests.m]: Yeah, it fails when I remove the method override

**Comment by ajkolean** [comment]
Any update on when this will be merged? Is there anything I can do to help?

**Comment by jlaws** [comment]
@d3mueller thanks for sending this out. @rnystrom Any chance you can review this again so it can be merged.  The build failures appear unrelated to the actual changes in the PR.
